src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128_1.cs

```diff
@@ -169,19 +181,50 @@ internal static void ThrowIfUnsupportedType()
         /// <param name="other">The <see cref="Vector128{T}" /> to compare with the current instance.</param>
         /// <returns><c>true</c> if <paramref name="other" /> is equal to the current instance; otherwise, <c>false</c>.</returns>
         /// <exception cref="NotSupportedException">The type of the current instance (<typeparamref name="T" />) is not supported.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public bool Equals(Vector128<T> other)
         {
             ThrowIfUnsupportedType();
 
-            for (int i = 0; i < Count; i++)
+            if (Sse.IsSupported && (typeof(T) == typeof(float)))
+            {
+                Vector128<float> result = Sse.CompareEqual(AsSingle(), other.AsSingle());
+                return Sse.MoveMask(result) == 0b1111; // We have one bit per element
```

Comment:
Originally, I thought that `Equals` may able to be optimized by `Sse41.Test*` to avoid the relatively high latency `MoveMask`. However, after calculating, I realized that the current implementation could be 1-cycle faster, as `cmp` could be macro-fused into the following jump-instruction if https://github.com/dotnet/coreclr/issues/17073 gets sovled. 

So, the current implementaion is good.

Reply:
Good to know. We might also want to or be able to special case comparison against all Zero or True (all bits set), but I think that would be separate.